Yoast/YoastSEO.js
# Add morphological analysis for English nouns

## Summary

This PR can be summarized in the following changelog entry:

* Implements morphological analysis for keyword nouns in English. 

## Relevant technical choices:

* For every English word build its possible singular and plural noun form using regexes and a list of exceptions. 
* The keyword filled in the focus keyword field is now checked for presence in the title in all its noun forms.

## Test instructions

This PR can be tested by following these steps:

* Add singular-plural forms of nouns to `/spec/morphology/englishNounsSpec.js` and check if `yarn test` passes
* Use browserified to check titleKeyword assessment. 
 * Single-word or multiple-word keyword: Try using different forms of nouns in the keyword field and in the title. Check if the assessment returns correct result.
* For multiple-word keyword, try using only function words inside keyword - the assessment should effectively search for them in the title. Try using both function and content words inside keyword - the assessment should only search for content words.
 * Try putting the keyword inside double quotation marks - the assessment should only search for exact matches in the title.

Fixes #1501. Fixes #1514
